api/registry/tasks.py

```diff
@@ -52,74 +31,30 @@ def score_registry_passport(community_id: int, address: str):
 
 
 def score_passport(community_id: int, address: str):
-    log.info(
-        "score_passport request for community_id=%s, address='%s'",
-        community_id,
-        address,
-    )
-
-    passport = None
-    try:
-        passport = load_passport_record(community_id, address)
-        if not passport:
-            log.info(
-                "Passport no passport found for address='%s', community_id='%s' that has requires_calculation=True or None",
-                address,
-                community_id,
-            )
-            return
-        remove_existing_stamps_from_db(passport)
-        passport_data = load_passport_data(address)
-        validate_and_save_stamps(passport, passport_data)
-        calculate_score(passport, community_id)
+    passport = load_passport_record(community_id, address)
 
-    except APIException as e:
-        log.error(
-            "APIException when handling passport submission. community_id=%s, address='%s'",
-            community_id,
+    if not passport:
+        log.info(
+            "Passport no passport found for address='%s', community_id='%s' that has requires_calculation=True or None",
             address,
-            exc_info=True,
-        )
-        if passport:
-            # Create a score with error status
-            Score.objects.update_or_create(
-                passport_id=passport.pk,
-                defaults=dict(
-                    score=None,
-                    status=Score.Status.ERROR,
-                    last_score_timestamp=None,
-                    evidence=None,
-                    error=e.detail,
-                ),
-            )
-    except Exception as e:
-        log.error(
-            "Error when handling passport submission. community_id=%s, address='%s'",
             community_id,
-            address,
-            exc_info=True,
         )
-        if passport:
-            # Create a score with error status
-            Score.objects.update_or_create(
-                passport_id=passport.pk,
-                defaults=dict(
-                    score=None,
-                    status=Score.Status.ERROR,
-                    last_score_timestamp=None,
-                    evidence=None,
-                    error=str(e),
-                ),
-            )
+        return
 
+    score, _ = Score.objects.get_or_create(
+        passport=passport,
+        defaults=dict(
+            score=None,
+            status=None,
+            last_score_timestamp=None,
+            evidence=None,
+            error=None,
+        ),
+    )
 
-def load_passport_data(address: str):
-    # Get the passport data from the blockchain or ceramic cache
-    passport_data = get_passport(address)
-    if not passport_data:
-        raise NoPassportException()
+    async_to_sync(ascore_passport)(passport.community, passport, address, score)
```

Comment:
This is awesome. I was worried about the code being defined in two places, this is a fantastic solution!